packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,96 +1,192 @@
+        const rect = el.getBoundingClientRect();
+        const { clientX, clientY } = e;
+
+        if (childElements?.length && clientX + 2 > rect.right && clientY > rect.top && clientY < rect.bottom) return;
 
-    const [open, setOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+        closeMenu();
+    };
 
-    const listClasses = [classes.root];
-    if (hasSelectedChild) listClasses.push(classes.childSelected);
-    if (open) listClasses.push(classes.open);
+    const listClasses = [];
+    if (hasSelectedChild.current) listClasses.push(classes.childSelected);
+    if (isSubmenuOpen) listClasses.push(classes.open);
 
     return (
-        <div {...otherProps} className={listClasses.join(" ")}>
-            <div className={classes.listItem}>
+        <div {...otherProps}>
+            <div
+                id="menu-item"
+                className={classes.listItem}
+                aria-haspopup="true"
+                aria-controls={isSubmenuOpen ? "mouse-over-menu" : undefined}
+                aria-expanded={isSubmenuOpen ? "true" : undefined}
+                onMouseOver={handlePopoverOpen}
+                onMouseLeave={handlePopoverClose}
+            >
                 <MenuItem
                     primary={primary}
                     secondary={secondary}
                     hasSubitems={!!childElements?.length}
-                    isCollapsibleOpen={open}
+                    isCollapsibleOpen={isSubmenuOpen}
+                    isMenuOpen={isMenuOpen}
                     icon={icon}
                     level={level}
-                    onClick={() => setOpen(!open)}
-                    secondaryAction={open ? openedIcon : closedIcon}
+                    onClick={() => setIsSubmenuOpen(!isSubmenuOpen)}
+                    secondaryAction={
+                        itemLevel === 1 && isSubmenuOpen ? (
+                            <OpenedIcon
+                                className={`${classes.collapsibleIcon} ${
+                                    !isMenuOpen && isSubmenuOpen && itemLevel === 1 ? classes.colorWhite : classes.colorGrey
+                                }`}
+                            />
+                        ) : (
+                            <ClosedIcon
+                                className={`${classes.collapsibleIcon} ${
+                                    !isMenuOpen && isSubmenuOpen && itemLevel === 1 ? classes.colorWhite : classes.colorGrey
+                                }`}
+                            />
+                        )
+                    }
+                    className={listClasses.join(" ")}
                 />
             </div>
-            <Collapse in={open} timeout="auto" unmountOnExit>
-                <List disablePadding>{childElements}</List>
-            </Collapse>
+            {!isMenuOpen && drawerVariant === "permanent" ? (
+                <Menu
+                    id="mouse-over-menu"
```

Comment:
The id seems very generic, which increases the chance of colliding with another components id.

Reply:
✅